core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1088,8 +1088,12 @@ abstract class RDD[T: ClassTag](
         partiallyAggregated = partiallyAggregated.mapPartitionsWithIndex {
           (i, iter) => iter.map((i % curNumPartitions, _))
         }.reduceByKey(new HashPartitioner(curNumPartitions), cleanCombOp).values
+        // This fails:
+        // .foldByKey(zeroValue, new HashPartitioner(curNumPartitions))(cleanCombOp).values
```

Comment:
If either of foldByKey (here) or fold (a few lines below) are used, then the test "treeAggregate with ops modifying first args" fails---but none other of the reduce/fold/aggregate tests fail.  Any ideas on why?